fdbserver/KeyValueStoreMemory.actor.cpp

```diff
@@ -427,7 +429,34 @@ class KeyValueStoreMemory : public IKeyValueStore, NonCopyable {
 		return log->push( LiteralStringRef("\x01") ); // Changes here should be reflected in OP_DISK_OVERHEAD
 	}
 
-	ACTOR static Future<Void> recover( KeyValueStoreMemory* self, bool exactRecovery ) {
+	// reset everything and join as an empty node
+	static void join_as_empty(KeyValueStoreMemory* self){
+		self->previousSnapshotEnd = -1;
+		self->currentSnapshotEnd = -1;
+		self->resetSnapshot = false;
+		self->committedWriteBytes = self->committedDataSize = 0;
+		self->transactionSize = 0;
+		self->transactionIsLarge = false;
+		// clear data set and operation queue
+		self->data.clear();
+		self->queue.clear();
+	}
+
+	// check if cluster is healthy, private method used by recovery process
+	ACTOR static Future<bool> is_cluster_healthy(){
+		// TODO: user can specify cluster file name
+		state std::pair<std::string, bool> resolvedClusterFile = ClusterConnectionFile::lookupClusterFileName("");
```

Comment:
hm, this is not really correct since the cluster file can be different for the process (and not "", which resolves to /etc/foundationdb/fdb.cluster by default on linux). There should be a way to do this over the network (to fetch the recovery status of the cluster. Please dig and ping me if you can't find it.